File: src/main/java/murphy/software/games/platform2d/move/movement/PositionController.java
package murphy.software.games.platform2d.move.movement;

import murphy.software.games.platform2d.move.model.Position;

public class PositionController {


    void move(Position mover, double xdiff, double ydiff, double zdiff) {
        mover.x(diff(mover.x(), xdiff));
        mover.y(diff(mover.y(), ydiff));
        mover.z(diff(mover.z(), zdiff));
    }

    private double diff(double d, double diff) {
        return d+diff;
    }
}

File: src/main/java/murphy/software/games/platform2d/move/movement/PositionController.kt
package murphy.software.games.platform2d.move.movement

import murphy.software.games.platform2d.move.model.Position2D

class PositionController {
    fun move(mover: Position2D, xdiff: Double, ydiff: Double) {
        mover.x = diff(mover.x, xdiff)
        mover.y = diff(mover.y, ydiff)
    }

    private fun diff(d: Double, diff: Double): Double {
        return d + diff
    }
}